domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/ActivityLifecycleObserverTest.kt

```diff
@@ -103,8 +102,13 @@ class ActivityLifecycleObserverTest {
   @Test
   fun testObserver_onFirstActivityResume_verifyCurrentScreenReturnsCorrectValue() {
     val activity = Robolectric.buildActivity(Activity::class.java).get()
-    activity.intent = currentAppScreenNameWrapper
-      .getCurrentAppScreenNameIntent(ScreenName.ACTIVITY_NAME_UNSPECIFIED)
+    activity.intent = Intent().apply {
+      this.putProtoExtra(
+        CurrentAppScreenNameIntentDecorator.getCurrentAppScreenNameIntentKey(),
+        CurrentAppScreenNameIntentDecorator
+          .decorateWithScreenName(ScreenName.ACTIVITY_NAME_UNSPECIFIED)
```

Comment:
Is this actually needed? I'd expect retrieval to default to unspecified if the screen name isn't present.

Reply:
Yes, but since we're decorating the activity's intent with a screen name -- like in case of an actual activity, we need to set an initial value. The getter will default to an unspecified value but I don't think the setter should take null as a value as of now as it'll defeat the entire purpose of adding screen name to the activity. That being said, do you think we should support a null case for the setter too? 

However, since it'll return an unspecified value even when null, I've modified the case to contain home_activity as the screen name.

Reply:
I'm not sure that I understand the 'null' part of the argument. If the proto is absent from the intent, the decorator will extract a default proto instance (which will always correspond to ACTIVITY_NAME_UNSPECIFIED).

I think having a specific name is better here, but perhaps it's worth testing this default scenario in the decorator's test suite.

Reply:
done.